File: src/main/model/entity/Resource.ts
import BaseEntity from '../../base/BaseEntity.js'
import { NotNullish } from '../../util/CommonUtil.js'

export default class Resource extends BaseEntity {
  /**
   * 作品id
   */
  worksId: number | undefined | null

  /**
   * 任务id
   */
  taskId: number | undefined | null

  /**
   * 状态（0：停用，1：启用）
   */
  state: number | undefined | null

  /**
   * 文件路径
   */
  filePath: string | undefined | null

  /**
   * 文件名
   */
  fileName: string | undefined | null

  /**
   * 扩展名
   */
  filenameExtension: string | undefined | null

  /**
   * 建议名称
   */
  suggestedName: string | undefined | null

  /**
   * 工作目录
   */
  workdir: string | undefined | null

  /**
   * 资源是否保存完成
   */
  resourceComplete: number | undefined | null

  /**
   * 导入方式（0：本地导入，1：站点下载）
   */
  importMethod: number | undefined | null

  constructor(resource?: Resource) {
    super(resource)
    if (NotNullish(resource)) {
      this.worksId = resource.worksId
      this.taskId = resource.taskId
      this.state = resource.state
      this.filePath = resource.filePath
      this.fileName = resource.fileName
      this.filenameExtension = resource.filenameExtension
      this.suggestedName = resource.suggestedName
      this.workdir = resource.workdir
      this.resourceComplete = resource.resourceComplete
      this.importMethod = resource.importMethod
    }
  }
}

File: src/main-go/internal/model/resource.go
package model

// Resource 资源
type Resource struct {
	ID                int64  `gorm:"primaryKey;column:id" json:"id"`
	WorkID            int64  `gorm:"column:work_id" json:"workId"`
	TaskID            int64  `gorm:"column:task_id" json:"taskId"`
	State             int    `gorm:"column:state" json:"state"`
	FilePath          string `gorm:"column:file_path" json:"filePath"`
	FileName          string `gorm:"column:file_name" json:"fileName"`
	FilenameExtension string `gorm:"column:filename_extension" json:"filenameExtension"`
	SuggestName       string `gorm:"column:suggest_name" json:"suggestName"`
	ResourceSize      int64  `gorm:"column:resource_size" json:"resourceSize"`
	Workdir           string `gorm:"column:workdir" json:"workdir"`
	ResourceComplete  int    `gorm:"column:resource_complete" json:"resourceComplete"`
	CreateTime        int64  `gorm:"column:create_time" json:"createTime"`
	UpdateTime        int64  `gorm:"column:update_time" json:"updateTime"`
}

func (Resource) TableName() string {
	return "resource"
}

func (e Resource) GetID() int64 {
	return e.ID
}
